src/free.rs

```diff
@@ -128,17 +128,24 @@ pub fn zip<I, J>(i: I, j: J) -> Zip<I::IntoIter, J::IntoIter>
     i.into_iter().zip(j)
 }
 
-/// Create an iterator that first iterates `i` and then `j`.
+/// Returns an iterator combining two iterators consecutively. 
```

Comment:
Let's just adapt the [documentation from std](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.chain): "Takes two iterables and creates a new iterator over both in sequence."

Reply:
Made suggested changes.